src/main/scala/no/gdl/bookapi/controller/FeaturedContentController.scala

```diff
@@ -0,0 +1,105 @@
+      summary "Returns all featured content in default language"
+      notes "Returns all featured content in default language"
+      parameters correlationIDParameter
+      responseMessages response500)
+
+    private val getAllFeaturedContentInLang = (apiOperation[Seq[api.Book]]("getAllFeaturedContentInLang")
+      summary "Returns all featured content in given language"
+      notes "Returns all featured content in given language"
+      parameters(correlationIDParameter,
+      pathParam[String]("lang").description("Desired language for books specified in ISO 639-1/2 format."))
+      responseMessages response500)
+
+    private val addNewFeaturedContent = (apiOperation[FeaturedContentId]("addNewFeaturedContent")
+      summary "Adds a new featured content"
+      notes "Returns id of created featured content"
+      parameters(correlationIDParameter, bodyParam[NewFeaturedContent]("JSON body"))
+      responseMessages response500)
+
+    private val updateExistingFeaturedContent = (apiOperation[FeaturedContentId]("updateExistingFeaturedContent")
+      summary "Updates an existing featured content"
+      notes "Returns id of updated featured content"
+      parameters(correlationIDParameter, bodyParam[FeaturedContent]("JSON body"))
+      responseMessages response500)
+
+    private val deleteFeaturedContent = (apiOperation[FeaturedContentId]("deleteFeaturedContent")
+      summary "Deletes an existing featured content"
+      notes "Returns 200 OK if featured content was deleted"
+      parameters(correlationIDParameter, pathParam[Long]("id").description("ID of featured content to delete"))
+      responseMessages response500)
+
+    get("/", operation(getAllFeaturedContentInDefaultLang)) {
+      readService.featuredContentForLanguage(LanguageTag(DefaultLanguage))
+    }
+
+    get("/:lang", operation(getAllFeaturedContentInLang)) {
+      val contentForLang = readService.featuredContentForLanguage(LanguageTag(params("lang")))
+      if (contentForLang.nonEmpty) contentForLang else readService.featuredContentForLanguage(LanguageTag(DefaultLanguage))
+    }
+
+    post("/", operation(addNewFeaturedContent)) {
+      assertHasRole(FeaturedContentAdminRole)
+      val newFeaturedContent = extract[NewFeaturedContent](request.body)
+      writeService.newFeaturedContent(newFeaturedContent) match {
+        case Success(x) => Created(x)
+        case Failure(ex) => throw ex
+      }
+    }
+
+    put("/", operation(updateExistingFeaturedContent)) {
+      assertHasRole(FeaturedContentAdminRole)
+      val updatedContent = extract[FeaturedContent](request.body)
+      writeService.updateFeaturedContent(updatedContent) match {
+        case Success(x) => Ok(x)
+        case Failure(ex) => throw ex
+      }
+    }
+
+    delete("/:id", operation(deleteFeaturedContent)) {
+      assertHasRole(FeaturedContentAdminRole)
+      val id = long("id")
+      writeService.deleteFeaturedContent(id) match {
+        case Success(_) => Ok(s"Deleted $id")
```

Comment:
Hmm...burde denne returnert en `204 No Content` i stedet for `200 OK`?

Reply:
Det er det delte meninger om, virker som om begge deler er ok.